examples/sp/sp_tutorial.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""A simple tutorial that shows some features of the Spatial Pooler."""
+
+import numpy as np
+import random
+import matplotlib.pyplot as plt
+from nupic.research.spatial_pooler import SpatialPooler as SP
+
+def pctOverlap(x1, x2, size):
+  """
+  Computes the percentage of overlap between vectors x1 and x2.
+  Parameters:
+  ----------
+  x1: binary vector whose size is specified in parameter size
+  x2: binary vector whose size is specified in parameter size
+  size: length of binary vectors
+  """
+  nonZeroX1 = np.count_nonzero(x1)
+  nonZeroX2 = np.count_nonzero(x2)
+  minX1X2 = min(nonZeroX1, nonZeroX2)
+  pctOverlap = 0
+  if minX1X2 > 0:
+    pctOverlap = float(np.dot(x1, x2))/float(minX1X2)
+  return pctOverlap
+
+
+
+def corruptVector(vector, noiseLevel):
+  """
+  Corrupts a binary vector by swaping its inner values depending
+  on the amount of noise specified by the user.
```

Comment:
May be clearer to write "Corrupts a binary vector by inverting noiseLevel percent of its bits."